### PR TITLE
Added Trie [Go]

### LINKED ISSUE #546: Trie [Go]
#### This issue is about [Trie](https://en.wikipedia.org/wiki/Trie).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by praffn**
Fixes #546 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by singhpratyush** [comment]
Hi! Thanks for showing interest in contributing to this repository.

Please make sure that you have ticked the points mentioned in PR description correctly.

Thanks again!

**Comment by singhpratyush** [comment]
Comment on 96766058b60b5a3ca998efa64fe5abd5067a8686, file trie/trie.go, line 7.

exported const AlphabetSize should have comment or be unexported

*Origin: GoLintBear, Section: `golint`.*

File: trie/trie.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	// we still got bytes left in the key.
+	// pick the next byte from the key
+	c := key[d]
+	// and get the next node at position c
+	// also increment d.
+	return get(node.next[c], key, d+1)
+}
+
+// put returns a pointer to a node (representing a subtrie)
+func put(nnode *node, key []byte, value string, d int) *node {
```
Comment (mohitkyadav) [inline trie/trie.go]: Please rename `nnode` and `d` variables to some informative names.

File: trie/trie.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+import (
+	"fmt"
+)
+
+// AlphabetSize is the size of the alphabet used.
+// All nodes will have an array of pointers to other nodes,
+// and this array will have the size of the alphabet.
+// Setting to a smaller size (in allowed alphabet) will
+// lower the space needed for the trie
+const AlphabetSize int = 256 // bytesize
```
Comment (mohitkyadav) [inline trie/trie.go]: Please avoid inline comments.

**Pull request merged**